### PR TITLE
[ENH, MAINT] MNE Analyze - Fix event bugs and add auto recompute of avg

### PR DESCRIPTION
**Pull request by gabrielbmotta**
- Fix issue in event plugin where list of trigger channels shows duplicates.
- Change EVENT_GROUP_UPDATED into EVENT_UPDATED and change when it is emitted to signal changes to events.
- Added setting to AveragingSettingsView to turn on autocompute. Changed 'compute' button to 'save'.
- Added auto re-compute averages to averaging plugin.
- Changed right click menu items to class members
- Added mutex to MainWindow to fix multiple threads trying to write to log simultaneously.

### CODE REVIEW

File: applications/mne_analyze/libs/anShared/Model/eventmodel.h [hunk 1]
```diff
@@ -323,7 +323,7 @@ class ANSHAREDSHARED_EXPORT EventModel : public AbstractModel
      *
      * @return Returns a matrix of formatted event data.
      */
-    MatrixXi getEventMatrix(int iGroup = 9999);
+    MatrixXi getEventMatrix();
```
Comment (juangpc) [inline applications/mne_analyze/libs/anShared/Model/eventmodel.h:326]: 👏🏻

File: applications/mne_analyze/libs/anShared/Model/eventmodel.cpp [hunk 2]
```diff
@@ -711,3 +703,88 @@ std::vector<uint> EventModel::getEventSelection() const
...
+void EventModel::getEventsFromNewData()
+{
+    auto info = m_pFiffModel->getFiffInfo();
+    auto raw = m_pFiffModel->getFiffIO()->m_qlistRaw.first().data();
+
+    auto previousLastSample = m_pFiffModel->getPreviousLastSample();
+
+    int iFirstSample = m_pFiffModel->absoluteFirstSample();
+
+    std::list<int> indexList;
```
Comment (juangpc) [inline applications/mne_analyze/libs/anShared/Model/eventmodel.cpp:729]: Could the name of the variable be stimChanList or something like that?

File: applications/mne_analyze/libs/anShared/Model/eventmodel.cpp [hunk 3]
```diff
@@ -711,3 +703,88 @@ std::vector<uint> EventModel::getEventSelection() const
...
+    for (int iChannelIndex : indexList){
+        QList<QPair<int,double>> detectedTriggerSamples = RTPROCESSINGLIB::detectTriggerFlanksMax(mSampleData,
+                                                                                                  iChannelIndex,
+                                                                                                  0,
+                                                                                                  m_dThreshold,
+                                                                                                  0);
+
+        QMap<double,QList<int>> mEventsinTypes;
+
+        for(QPair<int,double> pair : detectedTriggerSamples){
```
Comment (juangpc) [inline applications/mne_analyze/libs/anShared/Model/eventmodel.cpp:758]: reference would avoid creating a variable calling QPair constructor each iteration. 
const auto& would make it const so easier to optimize for the compiler?

**Comment by juangpc** [comment]
I really like it. Everything is so much cleaner. And one less macro! great.

**Review by juangpc — APPROVED** [review]

**Review by juangpc — APPROVED** [review]

**Comment by gabrielbmotta** [comment]
wait no this was merged

**Pull request merged**